go/libraries/doltcore/sqle/privileges/file_handler.go

```diff
@@ -81,12 +87,33 @@ func LoadPrivileges() ([]*grant_tables.User, []*grant_tables.RoleEdge, error) {
 	return data.Users, data.Roles, nil
 }
 
-var _ grant_tables.PersistCallback = SavePrivileges
+// LoadData reads the mysql.db file, returns nil if empty or not found
+func LoadData() (*serial.MySQLDb, error) {
+	fileMutex.Lock()
+	defer fileMutex.Unlock()
+
+	// TODO: right filepath?
+	buf, err := ioutil.ReadFile(mysqlDbFilePath)
+	if err != nil {
+		return nil, nil
+	}
+	if len(buf) == 0 {
+		return nil, nil
+	}
+
+	// TODO: Flat buffers?
+	mysqlDb := serial.GetRootAsMySQLDb(buf, 0)
+
+	return mysqlDb, nil
+}
+
+var _ mysql_db.PrivilegePersistCallback = SavePrivileges
+var _ mysql_db.DataPersistCallback = SaveData
 
-// SavePrivileges implements the interface grant_tables.PersistCallback. This is used to save privileges to disk. If the
+// SavePrivileges implements the interface mysql_db.PrivilegePersistCallback. This is used to save privileges to disk. If the
 // file path has not been previously set, this returns without error. This is so that the logic path can retain the
 // calls regardless of whether a user wants privileges to be loaded or persisted.
-func SavePrivileges(ctx *sql.Context, users []*grant_tables.User, roles []*grant_tables.RoleEdge) error {
+func SavePrivileges(ctx *sql.Context, users []*mysql_db.User, roles []*mysql_db.RoleEdge) error {
```

Comment:
Looking at this interface a bit more, I think the best way to do this is to have GMS actually handle all of the FlatBuffer logic. So we can have `SaveData` take a byte slice and write it to disk, and `LoadData` load the byte slice from a file and send it to GMS. We also would need that `LoadPrivilegesData` function in GMS as well.